### ISSUE TITLE
Blur isn't applied on background view until it's rendered onto screen

### ISSUE THREAD

**Issue by KiranPanesar** [kevin-hirsch/KVNProgress#18]
So I wanted to show a KVNProgress HUD as soon as a view controller's view appears on the screen, so the user can only see the blurred overlay.

Right now if you call ```[KVNProgress show]``` is anything other than the ```-viewDidAppear```, the HUD's background is totally transparent.

I would suggest listening to the view's properties to determine when its visibility changes and redraw the blur when that happens. Open to here other ideas.

**Comment by kevin-hirsch** [comment]
It is because it's a static blur based on a screenshot of the view. In your case, the view is not yet visible so this causes the issue you have. It's planned that I implement also a dynamic blur, so I'll keep you in touch from here when it's done ;)

**Comment by KiranPanesar** [comment]
Yeah, I thought that was the case. 

A possibility for iOS 8+ would be UIBlurEffect: 
https://developer.apple.com/library/prerelease/ios/documentation/UIKit/Reference/UIBlurEffect_Ref/index.html

**Comment by kevin-hirsch** [comment]
Yes, that what I was thinking for iOS 8, but I will have to find another solution for iOS 7 compatibility.

**Comment by KiranPanesar** [comment]
We have implemented UIBlurEffect for iOS 8 on our internal projects and we're exploring [GPUBlur](https://github.com/BradLarson/GPUImage) and other hardware accelerated options for iOS 7 support.

We're going to be working on that at the end of next week, so if we find a good hardware-accelerated live blur method for our projects, I will share it here!

**Comment by kevin-hirsch** [comment]
Thanks a lot! :+1:
